cleanlab/datalab/internal/adapter/imagelab.py

```diff
@@ -278,7 +367,68 @@ def find_issues(
             self.datalab.data_issues.collect_issues_from_imagelab(
                 self.imagelab, issue_types_copy.keys()
             )
-            if self.datalab.has_labels:
-                self.datalab._correlations_df = self.datalab._spurious_correlation()
         except Exception as e:
             print(f"Error in checking for image issues: {e}")
+
+        spurious_correlation_issue_types = (
+            SPURIOUS_CORRELATION_ISSUE["spurious_correlations"]
+            if issue_types is None
+            else issue_types.get(
+                "spurious_correlations", SPURIOUS_CORRELATION_ISSUE["spurious_correlations"]
+            )
+        )
+        spurious_correlation_threshold = spurious_correlation_issue_types.get(
+            "threshold", SPURIOUS_CORRELATION_ISSUE["spurious_correlations"]["threshold"]
+        )
+        try:
+            if self.datalab.has_labels:
+
+                spurious_correlation_handler = SpuriousCorrelationHandler(
+                    threshold=spurious_correlation_threshold
+                )
+                self.datalab.data_issues.info["spurious_correlations"] = (
+                    spurious_correlation_handler.handle_spurious_correlations(
+                        imagelab_issues=self.imagelab.issues,
+                        labels=self.datalab.labels,
+                    )
+                )
+        except Exception as e:
+            print(f"Error in checking for spurious correlations: {e}")
+
+
+class SpuriousCorrelationHandler:
+    def __init__(self, threshold: Optional[float] = None) -> None:
+        if threshold is None:
+            self.threshold = SPURIOUS_CORRELATION_ISSUE["spurious_correlation"]["threshold"]
+        else:
+            self._threshold = threshold
+
+    @property
+    def threshold(self) -> Optional[float]:
+        return self._threshold
+
+    @threshold.setter
+    def threshold(self, new_threshold: float) -> None:
+        self._threshold = new_threshold
```

Comment:
For the handler, there's not a strong reason to make `threshold` a property. Any threshold is passed from `lab.find_issues(..., issue_types={"spurious_correlations": {"threshold": ...}})` and into the constructor of this class. Remove the property getter and setter and just keep the constructor.

Also, avoid setting the default value `threshold = SPURIOUS_CORRELATION_ISSUE["..."]` in multiple places. Currently, it's being set within the constructor and in `ImagelabIssueFinderAdapter.find_issues`.

Reply:
I have updated the code and instead now use a method instead of a class.